tests/rptest/archival/s3_client.py

```diff
@@ -116,12 +126,15 @@ def populate_handler(filename, level):
                     populate_handler(h.baseFilename, h.level)
 
     def make_client(self):
-        cfg = Config(region_name=self._region,
-                     signature_version=self._signature_version,
-                     retries={
-                         'max_attempts': 10,
-                         'mode': 'adaptive'
-                     })
+        cfg = Config(
+            region_name=self._region,
+            signature_version=self._signature_version,
+            retries={
+                'max_attempts': 10,
+                'mode': 'adaptive'
+            },
+            s3={'addressing_style': f'{self._addressing_style}'},
+            use_fips_endpoint=True if self._use_fips_endpoint else None)
```

Comment:
fyi: I believe boto3 will automatically select the proper (virtual host) addressing scheme when `use_fips_endpoint` is set to `True`.  But you don't need to change anything.